src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java

```diff
@@ -410,8 +410,6 @@ private String extractSubject(SamlResponse samlResponse) throws Exception {
 
         if (samlRolesSeparator != null) {
             values = splitRoles(values);
-        } else {
-            values = trimRoles(values);
```

Comment:
Administritive roles shouldn't be trimmed. But what about other roles? Should all roles not be trimmed?

Reply:
Its all the roles which are coming in SAMLResponse, not just administrative. 
People facing issues with administrative as we only allow administrative in the service, but it is applicable to all the roles.

Reply:
This was as is from the very start. Probably some use case, but as said we should validated as is and not use trimming at all.

Reply:
I totally understand and agree that it should validate the roles itself not being trimmed. But looks like the `trimRoles()` here was introduced at the very beginning by [this commit](https://github.com/opensearch-project/security/commit/beef34004d22b232c5e309c1728b20e889e19adc). I'm actually interested in the reason why `trimRoles()` was added here at the very beginning. I hope we did not miss any use cases.

Reply:
The Sujith's commit was just a code transfer from old package to the opendistro security.

Its same way as when collaborated with SG.